src/pudl/transform/vceregen.py

```diff
@@ -0,0 +1,206 @@
+    """
+    logger.info(
+        "Preping Lat-Long-FIPS table for merging with the capacity factor tables"
+    )
+    # Get state name patterns from the POLITICAL_SUBDIVISIONS table
+    ps_usa_df = POLITICAL_SUBDIVISIONS[POLITICAL_SUBDIVISIONS["country_code"] == "USA"]
+    state_names = cleanstrings_snake(
+        ps_usa_df, ["subdivision_name"]
+    ).subdivision_name.tolist()
+    state_pattern = "|".join(state_names)
+    # Transform lat long fips table by making the county_state_names lowercase
+    # to match the values in the capacity factor tables. Fix FIPS codes with
+    # no leading zeros, and add a unique country and state field.
+    lat_long_fips = (
+        raw_vcegen__lat_lon_fips.pipe(simplify_columns)
+        .assign(county_state_names=lambda x: x.county_state_names.str.lower())
+        .assign(fips=lambda x: zero_pad_numeric_string(x.fips, 5))
+        .assign(
+            county=lambda x: x.county_state_names.str.extract(
+                rf"(.+)_({state_pattern})"
+            )[0]
+        )
+        .assign(
+            state=lambda x: x.county_state_names.str.extract(
+                rf"(.+)_({state_pattern})"
+            )[1]
+        )
+        .rename(
+            columns={
+                "lat_county": "latitude",
+                "long_county": "longitude",
+                "fips": "county_id_fips",
+            }
+        )
+    )
+
+    return lat_long_fips
+
+
+def _add_time_cols(df: pd.DataFrame) -> pd.DataFrame:
+    """Add datetime and year_hour columns.
+
+    This function adds a datetime column and a year_hour column.
+    The datetime column is important for merging the data with other
+    data, and the year_hour 0-8769 column is important for modeling
+    purposes. The report_year column is also helpful for filtering,
+    so we keep all three!
+
+    For leap years (2020), December 31st is excluded.
+
+    TO-DO: decide whether to have the year_hour start at 0 or 1.
+    and update the date_range accordingly.
+    """
+    logger.info("Adding datetime columns")
+    # This data is compiled for modeling purposes and skips the last
+    # day of a leap year. If we want to add a datetime column, we need
+    # to make sure that we skip the 31st of December, 2020.
+    # the [:-1] is because datetime will include both the 0th hour of the
+    # start date and the 0th hour of the end date. We only want one!
+    # Though we still need to decide which. This at least makes the math
+    # right.
+    to_leap_year_dates = pd.date_range(start="2019-01-01", end="2020-12-31", freq="h")[
+        :-1
+    ]
+    post_leap_year_dates = pd.date_range(
+        start="2021-01-01", end="2024-01-01", freq="h"
+    )[:-1]
+    # Add date sequence to pre and post leap year dates
+    df.loc[:, "report_year"] = df.report_year.astype("Int32")
+    df.loc[df["report_year"] < 2021, "hour"] = to_leap_year_dates
+    df.loc[df["report_year"] > 2020, "hour"] = post_leap_year_dates
```

Comment:
Highlighting this chunk of code because it's important to note! The leap year value that gets dropped is December 31st, 2020, not 2023. I hope this won't be an issue with datetime continuity, but it might @zaneselvans.

Reply:
Rather than treating the normal and leap years separately, you could define the date range using `periods=8760` rather than setting the `end` which would simplify the logic here, and work for any past or future year of 8760 data as well.

Reply:
It might look like:
```python
all_years = [2020, 2021]
datetime8760_index = pd.DatetimeIndex(
    pd.concat(
        [
            pd.Series(
                pd.date_range(start=f"{year}-01-01", periods=8760, freq="h")
            ) for year in all_years
        ]
    )
)
```

Reply:
oh cool, this is interesting to know that `periods` works like that.